public/Aframe/loading-screens/scene-1/components.js

```diff
@@ -0,0 +1,114 @@
+            e.detail.body.el; // Other entity, which playerEl touched.
+            e.detail.contact; // Stats about the collision (CANNON.ContactEquation).
+            e.detail.contact.ni; // Normal (direction) of the collision (CANNON.Vec3).
+        });
+    },
+});
+
+AFRAME.registerComponent("detect-button", {
+    init: function () {
+        var data = this.data;
+        var el = this.el;
+
+        el.addEventListener("mouseenter", function () {
+            el.setAttribute(
+                "animation__button-enter",
+                "property: scale; to: 1.1 1.1 1.1; dur: 500; dir:alternate; startEvents: mouseenter"
+            );
+        });
+
+        el.addEventListener("mouseleave", function () {
+            el.setAttribute(
+                "animation__button-leave",
+                "property: scale; to: 1 1 1; startEvents: mouseleave"
+            );
+        });
+
+        el.addEventListener("mousedown", function () {
+            el.setAttribute(
+                "animation__button-pressed",
+                "property: object3D.position.y; to: -5.5; dir: alternate; dur: 50; startEvents: mousedown"
+            );
+
+            el.emit("buttondown", false);
+        });
+
+        el.addEventListener("mouseup", function () {
+            el.setAttribute(
+                "animation__button-released",
+                "property: object3D.position.y; to: -5; dir: alternate; dur: 100; startEvents: mouseup"
+            );
+        });
+    },
+});
+
+function addBall() {
+    let container = document.querySelector("#container");
+    let colors = [
+        "red",
+        "orange",
+        "yellow",
+        "green",
+        "blue",
+        "purple",
+        "hotpink",
+    ];
+
+    let x = Math.random() * 10 - 5;
+    let y = Math.random() * 5 + 2;
+    let z = Math.random() * -10;
+
+    container.innerHTML += `<a-sphere collision-detect id="ball" click-drag dynamic-body position="${x} ${y} ${z}" radius="0.5" color="${
+        colors[Math.floor(Math.random() * colors.length)]
+    }" mass="0.5"></a-sphere>`;
+}
+
+AFRAME.registerComponent("add-ball", {
+    update: function () {
+        var data = this.data;
+        var el = this.el;
+
+        var curBalls = document.querySelectorAll("#ball");
+
+        for (let i = 0; i < curBalls.length; i++) {
+            let container = document.querySelector("#container");
+            container.removeChild(curBalls[i]);
+        }
+
+        el.addEventListener("buttondown", function () {
+            for (let i = 0; i < 10; i++) {
+                addBall();
+            }
+        });
+    },
+});
+
+AFRAME.registerComponent("gravity-switcher", {
+    /*init: function () { //not sure if this code is necessary but keeping it in just in case
+        var data = this.data;
+        var el = this.el; 
+    },*/
+
+    update: function () {
+        var data = this.data;
+        var el = this.el;
+
+        el.addEventListener("buttondown", function () {
```

Comment:
On line 83 and here on 101, we have eventlisteners with the same name, `buttondown`. Is that intentional?

Reply:
My intention at the time was to have the button both generate a ball and reverse the scene's gravity. So yes, that action was intentional at the time - although may change later (important to note that the scene doesn't update for some reason, even though I'm pretty sure I'm changing the attribute correctly. So it's probably an issue with the update() function)

Reply:
Oh I see, Thanks, was just curious